### ISSUE TITLE
npm-plugin can't execute tasks correctly on electron

### ISSUE THREAD

**Issue by owlJaeger** [eclipse-theia/theia#7737]
### Bug Description:
When trying to execute the `start`-script inside one of my projects, the `ShellTerminalServer` throws errors and the task doesn't execute how it should.
I've narrowed down the cause a bit. It has to do something with the backend-forking, happening when starting the electron version without `--no-cluster`. Using the `start:electron`-script of the theia monorepo will not use `backend-forking` und thus the issue can also be encountered when using that script. It should also happen to downstream projects. My project I've noticed this issue with, is a simple electron application. The start-script is `electron main.js`. I didn't encounter this issue while using vscode or a non-backend-forked version of theia. The error inside my project is:
```
app.whenReady().then(createWindow)
    ^

TypeError: Cannot read property 'whenReady' of undefined
    at Object.<anonymous> (c:\Users\Luca\Desktop\electron-quick-start\main.js:25:5)
    at Module._compile (internal/modules/cjs/loader.js:968:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:986:10)       
    at Module.load (internal/modules/cjs/loader.js:816:32)
    at Module._load (internal/modules/cjs/loader.js:728:14)
    at Module._load (electron/js2c/asar.js:717:26)
    at Function.Module._load (electron/js2c/asar.js:717:26)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1038:10)
    at internal/main/run_main_module.js:17:11
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! electron-quick-start@1.0.0 start: `electron .`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the electron-quick-start@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Luca\AppData\Roaming\npm-cache\_logs\2020-05-04T19_50_13_189Z-debug.log
```
The theia-log contains following multiple times:
```
root ERROR Request getCwdURI failed with error: terminal "3" does not exist Error: terminal "3" does not exist
    at ShellTerminalServer.<anonymous> (C:\Users\Luca\Desktop\theia\packages\terminal\lib\node\base-terminal-server.js:133:27)
    at step (C:\Users\Luca\Desktop\theia\packages\terminal\lib\node\base-terminal-server.js:59:23)
    at Object.next (C:\Users\Luca\Desktop\theia\packages\terminal\lib\node\base-terminal-server.js:40:53)
    at C:\Users\Luca\Desktop\theia\packages\terminal\lib\node\base-terminal-server.js:34:71
    at new Promise (<anonymous>)
    at __awaiter (C:\Users\Luca\Desktop\theia\packages\terminal\lib\node\base-terminal-server.js:30:12)
    at ShellTerminalServer.BaseTerminalServer.getCwdURI (C:\Users\Luca\Desktop\theia\packages\terminal\lib\node\base-terminal-server.js:128:16)
    at JsonRpcProxyFactory.<anonymous> (C:\Users\Luca\Desktop\theia\packages\core\lib\common\messaging\proxy-factory.js:218:73)
    at step (C:\Users\Luca\Desktop\theia\packages\core\lib\common\messaging\proxy-factory.js:47:23)
    at Object.next (C:\Users\Luca\Desktop\theia\packages\core\lib\common\messaging\proxy-factory.js:28:53)
```

### Steps to Reproduce:

1. Start theia without `--no-cluster` option ( executing the `start:electron` scripts will do this )
2. Opening a folder containing a simple electron application. ( Cloning the official [electron-quick-start](https://github.com/electron/electron-quick-start) repository and using this as the electron application will work )
3. Try to execute the `start`-script.

### Additional Information

- Operating System: Windows 10 (10.0.19041 Build 19041)
- Theia Version: 1.1.0 ( @theia/node-pty: 0.9.0-theia.6 )

**Comment by owlJaeger** [comment]
Seems like it isn't only related to the
```
root ERROR Request getCwdURI failed with error: terminal "3" does not exist Error: terminal "3" does not exist
    at ShellTerminalServer.<anonymous> (C:\Users\Luca\Desktop\theia\packages\terminal\lib\node\base-terminal-server.js:133:27)
    at step (C:\Users\Luca\Desktop\theia\packages\terminal\lib\node\base-terminal-server.js:59:23)
    at Object.next (C:\Users\Luca\Desktop\theia\packages\terminal\lib\node\base-terminal-server.js:40:53)
    at C:\Users\Luca\Desktop\theia\packages\terminal\lib\node\base-terminal-server.js:34:71
    at new Promise (<anonymous>)
    at __awaiter (C:\Users\Luca\Desktop\theia\packages\terminal\lib\node\base-terminal-server.js:30:12)
    at ShellTerminalServer.BaseTerminalServer.getCwdURI (C:\Users\Luca\Desktop\theia\packages\terminal\lib\node\base-terminal-server.js:128:16)
    at JsonRpcProxyFactory.<anonymous> (C:\Users\Luca\Desktop\theia\packages\core\lib\common\messaging\proxy-factory.js:218:73)
    at step (C:\Users\Luca\Desktop\theia\packages\core\lib\common\messaging\proxy-factory.js:47:23)
    at Object.next (C:\Users\Luca\Desktop\theia\packages\core\lib\common\messaging\proxy-factory.js:28:53)
```
error-message. When using the integrated terminal instead of the `npm scripts`-tasklauncher, no error-message is thrown on executing `npm start`. Creating a Terminal will give this error-message though:
```
root ERROR Error: Widget is not attached.
    at Function.detach (file:///C:/Users/Luca/Desktop/theia/examples/electron/lib/bundle.js:26807:19)
    at TerminalSearchWidget.../../node_modules/@phosphor/widgets/lib/widget.js.Widget.dispose (file:///C:/Users/Luca/Desktop/theia/examples/electron/lib/bundle.js:25945:20)
    at TerminalSearchWidget.../../packages/core/lib/browser/widgets/widget.js.BaseWidget.dispose (file:///C:/Users/Luca/Desktop/theia/examples/electron/lib/bundle.js:127169:34)
    at TerminalSearchWidget.disposable.dispose (file:///C:/Users/Luca/Desktop/theia/examples/electron/lib/bundle.js:128477:13)
    at DisposableCollection.../../packages/core/lib/common/disposable.js.DisposableCollection.dispose (file:///C:/Users/Luca/Desktop/theia/examples/electron/lib/bundle.js:128454:40)
    at TerminalWidgetImpl.../../packages/core/lib/browser/widgets/widget.js.BaseWidget.dispose (file:///C:/Users/Luca/Desktop/theia/examples/electron/lib/bundle.js:127170:24)
    at TerminalWidgetImpl.push.../../packages/terminal/lib/browser/terminal-widget-impl.js.TerminalWidgetImpl.dispose (file:///C:/Users/Luca/Desktop/theia/examples/electron/lib/35.bundle.js:1534:34)
    at TerminalWidgetImpl.../../packages/core/lib/browser/widgets/widget.js.BaseWidget.onCloseRequest (file:///C:/Users/Luca/Desktop/theia/examples/electron/lib/bundle.js:127174:14)
    at TerminalWidgetImpl.newValue.onCloseRequest (file:///C:/Users/Luca/Desktop/theia/examples/electron/lib/bundle.js:114889:43)
    at TerminalWidgetImpl.newValue.onCloseRequest (file:///C:/Users/Luca/Desktop/theia/examples/electron/lib/bundle.js:114889:43)
root ERROR Couldn't resize terminal -1, because it doesn't exist.
```

I can't really find the error with this... I've written an simple application with `@theia/node-pty` and forked that script from another file. Everything works fine with that. @akosyakov do you have any ideas where I should look?

**Comment by luca-schlecker** [comment]
I have found the error that's causing this problem. While forking, ELECTRON_RUN_AS_NODE` is set to 1. This is a bug fixed in newer versions. More about that bug [here](https://github.com/electron/electron/pull/19742). I've manually tested it by removing that flag and execution works as expected 👍 To fix this, electron has to be upgraded to at least 7.2.X or the flag would have to be removed manually.
